sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -472,23 +472,32 @@ class RapidsBufferCatalog(
   /**
    * Free memory in `store` by spilling buffers to the spill store synchronously.
    * @param store store to spill from
-   * @param targetTotalSize maximum total size of this store after spilling completes
+   * @param targetTotalSize optional maximum total size of this store after spilling
+   *                        completes. Only stores that want to stay within a certain size
+   *                        would calculate this target, other stores are going to spill
+   *                        anything and retry allocations.
    * @param stream CUDA stream to use or null for default stream
    * @return optionally number of bytes that were spilled, or None if this called
    *         made no attempt to spill due to a detected spill race
    */
   def synchronousSpill(
       store: RapidsBufferStore,
-      targetTotalSize: Long,
+      targetTotalSize: Option[Long] = None,
       stream: Cuda.Stream = Cuda.DEFAULT_STREAM): Option[Long] = {
     val spillStore = store.spillStore
     if (spillStore == null) {
       throw new OutOfMemoryError("Requested to spill without a spill store")
     }
-    require(targetTotalSize >= 0, s"Negative spill target size: $targetTotalSize")
-    logWarning(s"Targeting a ${store.name} size of $targetTotalSize. " +
-      s"Current total ${store.currentSize}. " +
-      s"Current spillable ${store.currentSpillableSize}")
+    targetTotalSize.map { tgt =>
```

Comment:
Nit: this should probably be using orElse..foreach rather than map..getOrElse since the result is not used.